src/main/java/com/example/cabbooking/servicetest/DriverService.java

```diff
@@ -27,19 +42,45 @@ public void addDriver(DriverDetailsVO driverDetailsVO) {
         LocationDetailsVO location = new LocationDetailsVO(locationDetailsVO.getXDistance(), locationDetailsVO.getYDistance());
         DriverDetailsVO driver = new DriverDetailsVO(driverDetailsVO.getName(), driverDetailsVO.getGender(), driverDetailsVO.getAge(), vehicle, location, true);
         drivers.put(driver.getName(), driver);
-        System.out.println(driver.getName()+" "+driver.getGender()+" "+driver.getAge()+" "+driver.isAvailable()+" "+driver.getLocation()+" "+driver.getVehicle());
+        System.out.println(driver.getName() + " " + driver.getGender() + " " + driver.getAge() + " " + driver.isAvailable() + " " + driver.getLocation() + " " + driver.getVehicle());
+    }
+
+    public void saveDriver(DriverDetailsVO driverDetailsVO) {
+
+        Vehicle vehicle = new Vehicle();
+        vehicle.setModel(driverDetailsVO.getVehicle().getModel());
+        vehicle.setRegistrationNumber(driverDetailsVO.getVehicle().getRegistrationNumber());
+
+        Location location = new Location();
+        location.setXDistance(driverDetailsVO.getLocation().getXDistance());
+        location.setYDistance(driverDetailsVO.getLocation().getYDistance());
+
+        Driver driver = new Driver();
+        driver.setName(driverDetailsVO.getName());
+        driver.setAge(driverDetailsVO.getAge());
+        driver.setGender(driverDetailsVO.getGender());
+        driver.setAvailable(driverDetailsVO.isAvailable());
+        driver.setVehicle(vehicle);
+        driver.setLocation(location);
+        //Eastablish connection between driver and location ,driver and vehcile.
+        vehicle.setDriver(driver);
+        //location.setDriver(driver);
+        driverRepository.save(driver);
+        vehicleRepository.save(vehicle);
+        locationRepository.save(location);
+
     }
 
-    public List<RideDetailsVO> findAvailableDrivers(LocationDetailsVO source,LocationDetailsVO destination, int maxDistance) {
+    public List<RideDetailsVO> findAvailableDrivers(LocationDetailsVO source, LocationDetailsVO destination, int maxDistance) {
         List<RideDetailsVO> availableDrivers = new ArrayList<>();
         for (DriverDetailsVO driver : drivers.values()) {
-            if (driver.isAvailable() && driver.getLocation().distanceTo(source)<= maxDistance) {
-                double fare = rideService.calculateFare(source,destination);
-                availableDrivers.add(new RideDetailsVO(driver,Math.round(fare*100.0)/100.0));
+            if (driver.isAvailable() && driver.getLocation().distanceTo(source) <= maxDistance) {
+                double fare = rideService.calculateFare(source, destination);
+                availableDrivers.add(new RideDetailsVO(driver, Math.round(fare * 100.0) / 100.0));
```

Comment:
can we seperate the logic here ? maybe 2 different methods ? one for finding the driver based on whatever business logic you have and another for calculating fare / price of trip , so the code will follow maintainability , easy to add more logic in future